### PR TITLE
feat: anonymous user chat handoff

### PR DESCRIPTION
**Pull request by bsadecki-magi**
Loom: https://www.loom.com/share/546ed965482a402ca5dc039f97d7afaa

# PR Description: Anonymous User Chat Handoff to Zendesk

## Overview
This PR adds support for anonymous users to transition to live agent support. Previously, only host-signed users could handoff to Zendesk. The implementation includes user creation, conversation management, and session handling for both authenticated and anonymous users.

## Key Changes
- Added anonymous Zendesk user creation with unique ID format: `maven-anonymous-user-{nanoid}`
- Added email collection form for anonymous users during handoff
- Enhanced conversation management to track authenticated vs anonymous sessions
- Added test coverage for anonymous user scenarios
- Added error handling and validation for handoff requests
- Added a convention where actions with name `escalate-live-agent` use a custom action response form

## Technical Details
- Anonymous Zendesk users get distinct IDs using nanoid
- User creation logic branches between authenticated/anonymous paths
- Handoff tokens include isAuthenticated flag
- Added system message for anonymous handoffs to notify agents
- Moved handoff status enum to constants file
- Added AuthProvider context to manage user state

## Testing Coverage
New test cases for:
- Anonymous user creation flows
- Email collection form validation
- Handoff token generation
- Error handling for missing credentials
- Integration with chat components

### 1. Anonymous User Chat
1. Open chat widget without signed user data: https://chat-develop.onmaven.app/demo/monarch/godzilla?anonymous
2. Send a message
3. Click "Connect to Live Agent"
4. Enter email address
5. Verify connection to live agent: https://d3v-mavenagi.zendesk.com/agent
6. Verify chat history persists

### 2. Form Validation
1. Submit handoff form without email
2. Verify error message appears
3. Submit with valid email
4. Verify handoff initializes

### 3. Session Management
1. Start anonymous chat
2. Connect to agent
3. Exchange messages
4. End chat
5. Verify chat state resets

### 4. Error Cases
1. Try handoff with missing config
2. Check error messages
3. Test network failures
4. Verify error handling

### CODE REVIEW

File: app/[orgFriendlyId]/[id]/page.tsx [hunk 1]
```diff
@@ -25,20 +27,11 @@ import type {
 import type { Message } from "@/types";
 import type { Front } from "@/types/front";
 
-function ChatPage({ signedUserData }: { signedUserData: string | null }) {
+function ChatPage() {
```
Comment (bsadecki-magi) [inline app/[orgFriendlyId]/[id]/page.tsx:30]: Moved `signedUserData` to the new `AuthProvider`. No more manually passing props down to ancestor components.

File: app/[orgFriendlyId]/[id]/page.tsx [hunk 2]
```diff
@@ -69,22 +60,27 @@ function ChatPage({ signedUserData }: { signedUserData: string | null }) {
...
+    | Message
+    | ZendeskWebhookMessage
+    | ChatEstablishedMessage
+    | ChatEndedMessage
+    | Front.WebhookMessage
+  )[] = useMemo(() => {
+    return [...messages, ...handoffChatEvents]
+      .filter(({ timestamp }) => !!timestamp)
+      .sort((a, b) => (a.timestamp || 0) - (b.timestamp || 0));
+  }, [messages, handoffChatEvents]);
```
Comment (bsadecki-magi) [inline app/[orgFriendlyId]/[id]/page.tsx:80]: The previous implementation had a nasty recursion habit.

File: app/api/create/route.ts [hunk 3]
```diff
@@ -169,27 +168,41 @@ export async function POST(req: NextRequest) {
...
               controller.close();
             } catch (err) {
               console.error("Stream error:", err);
               controller.error(err);
             }
           },
+
+          cancel() {
+            isControllerClosed = true;
+          },
```
Comment (bsadecki-magi) [inline app/api/create/route.ts:205]: Added a `cancel` controller to break us from the loop.

File: app/api/zendesk/conversations/route.ts [hunk 4]
```diff
@@ -80,28 +89,70 @@ const getOrCreateZendeskConversation = async (
...
+will need to confirm the user's identity before continuing.
+
+Claiming email address: ${email}`,
+  },
+});
+
 export async function POST(req: NextRequest) {
   return withAppSettings(req, async (request, settings) => {
-    const { messages, signedUserData } = await request.json();
+    const { messages, signedUserData, email } = await request.json();
```
Comment (bsadecki-magi) [inline app/api/zendesk/conversations/route.ts:109]: `signedUserData` = authenticated
`email` = anonymous (we use the email in a message to the Zendesk agent.)

File: app/api/zendesk/conversations/route.ts [hunk 5]
```diff
@@ -117,6 +168,11 @@ export async function POST(req: NextRequest) {
 
     const { id: conversationId } = conversation;
 
+    const isAuthenticated = !!verifiedUserInfo;
+    if (!isAuthenticated) {
+      messages.push(createUnauthenticatedMessage(userId, email));
+    }
+
```
Comment (bsadecki-magi) [inline app/api/zendesk/conversations/route.ts:175]: Let the Zendesk agent know this user is not authenticated and they're claiming to be {email}

File: app/api/zendesk/conversations/route.ts [hunk 6]
```diff
@@ -126,7 +182,7 @@ export async function POST(req: NextRequest) {
     );
 
     const token = jwt.sign(
-      { scope: "appUser", userId, conversationId },
+      { scope: "appUser", userId, conversationId, isAuthenticated },
```
Comment (bsadecki-magi) [inline app/api/zendesk/conversations/route.ts:185]: Provide the client with a signed `isAuthenticated` property. We don't do anything with it yet.

File: app/demo/[orgFriendlyId]/[agentFriendlyId]/page.tsx [hunk 7]
```diff
@@ -15,9 +15,12 @@ const mockUserData = {
 
 export default async function Page({
   params,
+  searchParams,
```
Comment (bsadecki-magi) [inline app/demo/[orgFriendlyId]/[agentFriendlyId]/page.tsx:18]: Add a feature to the `/demo/org/agent` page: Simulate an anonymous user if `?anonymous` added to URL.

File: lib/useHandoff.ts [hunk 8]
```diff
@@ -250,31 +241,39 @@ export function useHandoff({
...
+        {
+          method: "POST",
+          body: JSON.stringify({
+            messages: handoffMessages,
+            signedUserData,
+            email,
+          }),
+          headers: generatedHeaders,
+        },
+      );
```
Comment (bsadecki-magi) [inline lib/useHandoff.ts:261]: Add `email` to request body.

File: lib/useHandoff.ts [hunk 9]
```diff
@@ -250,31 +241,39 @@ export function useHandoff({
...
 
-    const handoffAuthToken = response.headers.get(HANDOFF_AUTH_HEADER);
+      if (!response.ok) {
+        throw new Error(`HTTP error! status: ${response.status}`);
+      }
```
Comment (bsadecki-magi) [inline lib/useHandoff.ts:265]: The error here will bubble up to the `EscalationFormDisplay` component and display an error to the user.

**Review by igorien2k — APPROVED** [review]

**Pull request merged**